### PR TITLE
Fix broken status badge in README: nightly run CI in main branch

### PR DESCRIPTION
**Pull request by jcwchen**
### Description

Fix broken status badge in README: running CI nightly in main branch from Azurepipelines

### Motivation and Context

Fixes https://github.com/onnx/onnx/issues/5577.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,9 +6,9 @@ Copyright (c) ONNX Project Contributors
 
 <p align="center"><img width="40%" src="https://github.com/onnx/onnx/raw/main/docs/onnx-horizontal-color.png" /></p>
 
-[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status/Windows-CI?branchName=main&label=Windows)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=5&branchName=main)
-[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status/Linux-CI?branchName=main&label=Linux)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=7&branchName=main)
-[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status/MacOS-CI?branchName=main&label=MacOS)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=6&branchName=main)
+[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status%2FWindows-CI?branchName=refs%2Fpull%2F5568%2Fmerge)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=5&branchName=refs%2Fpull%2F5568%2Fmerge)
```
Comment (justinchuby) [inline README.md:9]:
```suggestion
[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status%2FWindows-CI?branchName=refs%2Fpull%2F5568%2Fmerge)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=5&branchName=refs%2Fpull%2F5568%2Fmerge&label=Windows)
```

try setting the label?

Comment (justinchuby) [reply README.md]: should we display results from pulls?

File: README.md [hunk 2]
```diff
@@ -6,9 +6,9 @@ Copyright (c) ONNX Project Contributors
 
 <p align="center"><img width="40%" src="https://github.com/onnx/onnx/raw/main/docs/onnx-horizontal-color.png" /></p>
 
-[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status/Windows-CI?branchName=main&label=Windows)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=5&branchName=main)
-[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status/Linux-CI?branchName=main&label=Linux)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=7&branchName=main)
-[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status/MacOS-CI?branchName=main&label=MacOS)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=6&branchName=main)
+[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status%2FWindows-CI?branchName=refs%2Fpull%2F5568%2Fmerge)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=5&branchName=refs%2Fpull%2F5568%2Fmerge&label=Windows)
```
Comment (justinchuby) [inline README.md:9]: Sorry wrong spot:
```suggestion
[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status%2FWindows-CI?branchName=refs%2Fpull%2F5568%2Fmerge&label=Windows)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=5&branchName=refs%2Fpull%2F5568%2Fmerge)
```

Comment (jcwchen) [inline README.md]:
```suggestion
[![Build Status](https://dev.azure.com/onnx-pipelines/onnx/_apis/build/status%2FWindows-CI?branchName=refs%2Fpull%2F5568%2Fmerge&label=Windows)](https://dev.azure.com/onnx-pipelines/onnx/_build/latest?definitionId=5&branchName=refs%2Fpull%2F5568%2Fmerge)
```

File: README.md (hunk 1 above)
Comment (jcwchen) [reply README.md]: > try setting the label?

Good catch. Updated three of them.

> should we display results from pulls? Maybe not?

I think only showing the result from the main branch seems clearer.

**Comment by justinchuby** [comment]
Should we include only prs on branches with name `gh-readonly-queue/main/*`?

**Comment by jcwchen** [comment]
> Should we include only pushes to main? Otherwise all PR status will influence the badge.

I got your point now. Previously I removed main branch trigger due to saving CI bandwidth: https://github.com/onnx/onnx/pull/5432 and I guess that's the reason now README shows never built. If we want the badge to show the normal status for the main branch, I think we will need to add main branch trigger back. The disadvantage is it will enhance CI loading.

BTW, we cannot do the same thing for `gh-readonly-queue/**` branches, because the branch name is dynamic and it seems we can only provide a static branch name in badge API: https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/list?view=azure-devops-rest-7.0.

**Comment by justinchuby** [comment]
I see. Maybe run the pipelines on main periodically and report that then?

**Review by justinchuby — APPROVED** [review]

File: .azure-pipelines/Linux-CI.yml
Comment (justinchuby) [inline .azure-pipelines/Linux-CI.yml:1]: Blocking on merges for now

**Comment by jcwchen** [comment]
> I see. Maybe run the pipelines on main periodically and report that then? We don't need them to run on every merge

Are you saying automatically or manually run them periodically? Still, either way seems unnecessarily complex to me. Could we just re-enable AzurePipelines to run in the main branch for now? We can come back to review this solution if this PR really brings traffic again. Nowadays onnx doesn't have much traffic unless there are a lot of dependency updates per week.

Also, I hope in the near future the merge queue feature should provide an option to specify required CI from their end. Now actually it makes no sense that the same CIs need to run twice in the main branch and the merge queue. If this enhancement has been implemented, it can solve our traffic issue here.

**Comment by justinchuby** [comment]
> Are you saying automatically or manually run them periodically? Still, either way seems unnecessarily complex to me. Could we just re-enable AzurePipelines to run in the main branch for now? We can come back to review this solution if this PR really brings traffic again. Nowadays onnx doesn't have much traffic unless there are a lot of dependency updates per week.

I mean an automatic trigger:

```yaml
schedules:
- cron: '0 0 * * *'
  displayName: Periodic CI
  branches:
    include:
    - main
```

I am fine either way. Thanks!

**Comment by jcwchen** [comment]
@justinchuby that way is clear and simple. Thank you for the suggestion! I have set running main branch nightly. PTAL.

**Review by justinchuby — APPROVED** [review]

**Pull request merged**